### ISSUE TITLE
Multiple Workers = Lost jobs?

### ISSUE THREAD

**Issue by roynasser** [chrisboulton/php-resque#32]
As per a comment on issue #30, I'm opening this issue in order to help gather more info and find a fix for the bug.

I observed that when deploying multiple workers, some jobs dont get processed...

Basically the test I did was:

watch output and number of jobs, launch 5 workers.
Enqueue 10 jobs.  number of jobs in redis gets augmented to 10 (so 10 jobs queued), but only a few get really processed.  Nonetheless, they all get pulled out of the queue...

So basically it is "loosing" jobs.

I will try and test more tomorrow, time permitting, and post back.

Chris said he had already noticed this, so if anyone else has noticed and/or fixed this, please let us know.

Imo this is crucial for any production environment, no?  Is anyone using phpresque in a production environment? Are you loosing jobs? Maybe you havent noticed?  Can you please check and revert?  

thanks!

**Comment by Cyruxx** [comment]
This seems to be still an issue, we're getting this behavior too.

**Comment by Cyruxx** [comment]
So it seems like there must be a bug within Redisent, as soon as we replace that with Predis everything works totally fine.

**Comment by danhunsaker** [comment]
Given we haven't actually used Redisent for quite some time, that's hardly surprising.  Credis does seem to still have some issues inherited from Redisent, though, especially at the version we've been locked to of late.  Predis seems likely to be even more stable than that.  Which is why switching to Predis (or, better, to be Redis-library-agnostic) is such a high priority.

**Comment by Cyruxx** [comment]
@danhunsaker I'm actually right now migrating everything to redis and PSR-4

**Comment by danhunsaker** [comment]
I take it by "everything" you mean this entire library.  Which ... seems like a decent amount of work.  Starting from `master`, or the latest tag?

**Comment by Cyruxx** [comment]
The latest tag actually, and I'm making progress see my fork :-)

**Comment by danhunsaker** [comment]
Oh.  Uh, don't do that.  That's a terrible idea.  The latest tag is several years - and several bugfixes, many of them major ones - old.  You definitely wanna start from `master`.

**Comment by Cyruxx** [comment]
@danhunsaker So basically everyone that check out the current project via composer is using a very  old version?

**Comment by danhunsaker** [comment]
Yes.  As has been discussed in countless other issues opened in the time since then, the latest tag is quite old, and quite a lot of commits have been made since, but despite fixing a great number of bugs and adding or enhancing a few features, no new tags have been added.  Momentum has picked up again of late, but the library's not *quite* to a new tagging point yet.  It will be, soon.  In the meantime, users are, unfortunately, forced to require the `dev-master` version, or the ID of the specific git commit they're working against.  It's not ideal, but it's the situation we're in at the moment.

**Comment by yccheok** [comment]
@amahrt Recently, we stuck at old 1.2 tag, and old PHP 5.3.3 and hit by this bug on high traffic server.

We are looking for possible a quick hot fix.

May I know, currently, do you use the following code in any of production - https://github.com/amahrt/php-resque/commits/bugfix/predis ?

**Comment by mbdwey** [comment]
+1 
ENV : docker [redis server,2 workers]
each worker got 50 fork and after testing 1000 task lose between 200 : 400 task each run
